src/databricks/labs/ucx/hive_metastore/table_migrate.py

```diff
@@ -594,3 +599,29 @@ def _sql_alter_from(self, table: Table, target_table_key: str, ws_id: int):
             f"('upgraded_from' = '{source}'"
             f" , '{table.UPGRADED_FROM_WS_PARAM}' = '{ws_id}');"
         )
+
+
+class TableMigrationOwnership(Ownership[TableMigrationStatus]):
+    """Determine ownership of table migration records in the inventory.
+
+    This is the owner of the source table, if (and only if) the source table is present in the inventory.
+    """
+
+    def __init__(self, tables_crawler: TablesCrawler, table_ownership: TableOwnership) -> None:
+        super().__init__(table_ownership._administrator_locator)  # TODO: Fix this
```

Comment:
```suggestion
    def __init__(self, administrator_locator: AdministratorLocator, tables_crawler: TablesCrawler, table_ownership: TableOwnership) -> None:
        super().__init__(administrator_locator)
```
don't do indirect dependencies.